tests/integration/expired_client.go

```diff
@@ -285,10 +285,11 @@ func upgradeExpiredClient(s *CCVTestSuite, clientTo ChainType) {
 	tmClientState.AllowUpdateAfterExpiry = true
 	hostChain.App.GetIBCKeeper().ClientKeeper.SetClientState(hostChain.GetContext(), substitute, tmClientState)
 
-	content := clienttypes.NewClientUpdateProposal(ibctesting.Title, ibctesting.Description, subject, substitute)
+	recoverMsg := clienttypes.NewMsgRecoverClient(hostChain.App.GetIBCKeeper().GetAuthority(), subject, substitute)
+	err = recoverMsg.ValidateBasic()
+	s.Require().NoError(err)
 
-	updateProp, ok := content.(*clienttypes.ClientUpdateProposal)
-	s.Require().True(ok)
-	err = hostChain.App.GetIBCKeeper().ClientKeeper.ClientUpdateProposal(hostChain.GetContext(), updateProp)
+	res, err := hostChain.App.GetIBCKeeper().RecoverClient(hostChain.GetContext(), recoverMsg)
 	s.Require().NoError(err)
+	s.Require().NotNil(res)
```

Comment:
isn't this covered by the NoError check? don't think it's needed

Reply:
removed